### ISSUE TITLE
Modal inside component

### ISSUE THREAD

**Issue by technbuzz** [vexip-ui/vexip-ui#374]
### Description

I have added below code.
```
  <Modal
    v-model:active="newModel"
    title="نیہ تدریج"
    :width="350"
    top="auto"
    transfer
    no-footer
  >
    <AddPersonForm :id="editablePersonId" />
  </Modal>
  
  // AddPerson
  <script setup lang="ts">
    import { onMounted, reactive, ref, watch } from 'vue';
    onMounted(() => {
      console.log('im rendered')
    })
</script>
```
This always logs the `im rendered` even the component is not visible. What is the best way to load component only when modal is open? (I am somewhat new to vue)

### Validations

- [X] Ensure this issue neither a bug report nor a feature proposal.
- [X] Read the [Contributing Guidelines](https://github.com/vexip-ui/vexip-ui/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://www.vexipui.com/).
- [X] Check that there isn't [already an issue](https://github.com/vexip-ui/vexip-ui/issues) that descript the same thing to avoid creating a duplicate.

**Comment by qmhc** [comment]
You can use the `show` event, which is emitted after the opening transition end.

**Comment by technbuzz** [comment]
where should I use the `show` property, can you share some code snippet?

**Comment by qmhc** [comment]
Are you want the `AddPersonForm` loads after `Modal` opened?

Or you want to remove the `Modal` from dom tree when it's not active?

**Comment by technbuzz** [comment]
I want to add `AddPersonFrom` to the `DOM` after `Modal` open.

**Comment by technbuzz** [comment]
This way helped me as well
```
<template>
  <Modal
        v-model:active="opened"
  >
    <AddPersonForm v-if="opened" :id="editablePersonId" />
  </Modal>
</template>
  
<script setup lang="ts">
import { ref } from 'vue'

const opened = ref(false)
</script>
```

One question, there is now way to give a reference of component just like *Angular Material* does?
Link the [example](https://material.angular.io/components/dialog/overview#dialog-overview)
```
openDialog(): void {
    const dialogRef = this.dialog.open(DialogOverviewExampleDialog, {
      data: {name: this.name, animal: this.animal},
    });

    dialogRef.afterClosed().subscribe(result => {
      console.log('The dialog was closed');
      this.animal = result;
    });
  }
```

**Comment by qmhc** [comment]
Unfortunately, there is no such way now. Maybe future will.

**Comment by qmhc** [comment]
I thought of that maybe you can use `tsx` to implement the way.

**Comment by qmhc** [comment]
Now provide a hook method `useModal` to create a disposable modal via js way. You can view the Modal docs and it will be released in `2.2.3`.
